bikes_new.sql

```diff
@@ -0,0 +1,142 @@
+
+/*weekendy*/
+create view v_weekendy as
+SELECT distinct time, extract(year from time)||'-'||extract(month from time)||'-'||extract(day from time) as nowa_data
+  FROM status_csv sc 
+ WHERE EXTRACT(ISODOW FROM sc.time) IN (6, 7)
+ 
+ /*dni pracujace*/
+create view v_dni_robocze as
+SELECT distinct time, extract(year from time)||'-'||extract(month from time)||'-'||extract(day from time) as nowa_data
+  FROM status_csv sc 
+ WHERE EXTRACT(ISODOW FROM sc.time) IN (1, 2,3,4,5)
+ 
+ /*ekstrakcja godziny *//
+ select extract(hour from time) as godzina
+ from status_csv sc 
+
+ /*podzial na pory dnia*/
+create view v_pora_dnia1 as
+SELECT time, CASE 
+        WHEN h >= 0  AND h <= 7  THEN 'dawn'  -- including 0 !
+        WHEN h >  7  AND h <= 12 THEN 'morning'
+        WHEN h > 12  AND h <= 20 THEN 'evening'
+        WHEN h > 20              THEN 'night'
+       END AS moment
+FROM  (
+    SELECT time, extract(hour from time) AS h
+    FROM   status_csv sc 
+    ) x
+
+/*srednia, min i max rowerow po stacji i dniu*/
+select 
+sc2.station_id, 
+avg(sc2.bikes_available),
+min(sc2.bikes_available),max(sc2.bikes_available),
+extract(year from time)||'-'||extract(month from time)||'-'||extract(day from time) as nowa_data
+from status_csv sc2 
+group by sc2.station_id, extract(year from time)||'-'||extract(month from time)||'-'||extract(day from time)
+
+/*srednia, min i max rowerow po stacji i w weekendy*/
+select *,
+avg(sc.bikes_available) over (partition by nowa_data) as sr_w_weekend,
+min(sc.bikes_available) over (partition by nowa_data) as min_w_weekend,
+max(sc.bikes_available) over (partition by nowa_data) as max_w_weekend
+from status_csv sc 
+join v_weekendy 
+on v_weekendy.time=sc.time
+
+
+
+/*srednia, min i max rowerow po stacji i w dni pracujace*/
+
+select *,
+avg(sc.bikes_available) over (partition by  nowa_data) as sr_w_drobocze,
+min(sc.bikes_available) over (partition by  nowa_data) as min_w_drobocze,
+max(sc.bikes_available) over (partition by nowa_data) as max_w_drobocze
+from status_csv sc 
+join v_dni_robocze
+on v_dni_robocze.time=sc.time
+
+
+
+/*polaczenie z tripami Mateusza */
+--na miesiace
+select distinct v_top10a.trip_id,
+extract(year from sc2.time)||'-'||extract(month from sc2.time) as rok_miesiac,
```

Comment:
tu można byłoby też użyć funkcji date_trunc('month', sc2.time). Ale to rozwiązanie też jest poprawne